lib/tic_tac_toe/display.ex

```diff
@@ -0,0 +1,6 @@
+defmodule TicTacToe.Display do
+  def output(message, messages \\ []) do
+    IO.puts(message)
```

Comment:
How could we test that this calls `IO.puts`? How would we do this in an OO language?

Reply:
You could pass in `IO`, and you could even make that a default

```elixir
def output(message, messages \\ [], io \\ IO)
```

Alternatively, you could just pass in the function itself

```elixir
def output(message, messages \\ [], puts \\ IO.puts/1)
  puts.(messages)
  # ...
end
```

Lastly, you could turn the last two arguments into a "opts" Keyword list that makes this a little more flexible

```elixir
def output(messages, opts \\ []) do
  callback = Keyword.get(opts, :puts, IO.puts/1)
  messages = Keyword.get(opts, :messages, [])

  callback.(message)
  [message | messages]
end
```

Reply:
I would inject IO into the initialize method.  But here, I would have to pass as an argument, which seems unmaintainable.  Is there another way?

Reply:
This would allow you to inject a different version in test so you could verify the behavior. In the production code, you could use default.

Reply:
Got it. I included just io \\ IO in the argument, and removed the messages bit. I don't need this stateful object for this iteration.